src/osdag/design_type/compression_member/Column.py

```diff
@@ -865,19 +873,33 @@ def design_column(self):
         #     self.design_status = False
         #     self.design_status_list.append(self.design_status)
         self.epsilon = math.sqrt(250 / self.material_property.fy)
-        if len(self.input_section_list) > 0:
+        #if len(self.input_section_list) > 0:
+
+        # initializing lists to store the optimum results based on optimum UR and cost
+
+        # 1- Based on optimum UR
+        self.optimum_section_ur_results = {}
+        self.optimum_section_ur = []
 
-            # initializing lists to store the optimum results based on optimum UR and cost
+        # 2 - Based on optimum cost
+        self.optimum_section_cost_results = {}
+        self.optimum_section_cost = []
+        self.flag = self.section_classification(self)
 
-            # 1- Based on optimum UR
-            self.optimum_section_ur_results = {}
-            self.optimum_section_ur = []
+        print('self.flag:',self.flag)
+        # reduction of the area based on the connection requirements (input from design preferences)
+        if self.effective_area_factor < 1.0:
+            self.effective_area = round(self.effective_area * self.effective_area_factor, 2)
 
-            # 2 - Based on optimum cost
-            self.optimum_section_cost_results = {}
-            self.optimum_section_cost = []
+            #logger.warning("Reducing the effective sectional area as per the definition in the Design Preferences tab.")
+            #logger.info("The actual effective area is {} mm2 and the reduced effective area is {} mm2 [Reference: Cl. 7.3.2, IS 800:2007]".
+            #            format(round((self.effective_area / self.effective_area_factor), 2), self.effective_area))
+        #else:
+            #if self.section_class != 'Slender':
+            #    logger.info("The effective sectional area is taken as 100% of the cross-sectional area [Reference: Cl. 7.3.2, IS 800:2007].")
```

Comment:
This commented code is part of addition. Is it really needed? If yes, write a TODO.

Reply:
Commented out for the same reason, as they have been mentioned later to only append for optimum section to avoid clutter in the logger in case of all sections. However, later when other section profiles are added, it will help in debugging. Should I add a TODO and submit them again?